src/OpenProject/WebViewIntegration/OpenProjectInstanceValidator.cs

```diff
@@ -93,11 +80,9 @@ string appendSuffix(string uri)
       return null;
     }
 
-    private static async Task<HttpResponseMessage> GetHttpResponseAsync(string instanceUrl)
+    private async Task<HttpResponseMessage> GetHttpResponseAsync(string instanceUrl)
     {
-      using var httpClient = _services
-        .GetRequiredService<IHttpClientFactory>()
-        .CreateClient(nameof(OpenProjectInstanceValidator));
+      using var httpClient = _httpClientFactory.CreateClient(nameof(OpenProjectInstanceValidator));
```

Comment:
I'm not sure if we should be using the dispose pattern with `using` here if we get the client from the `HttpClientFactory`. I just don't know😀

Reply:
`using` should be the right way to access the client here. it is just a bit of emphasising for the GC that the client can be disposed right after the context was left. (maybe not only for the GC but also for the developer reading this peace of code ;) )

Reply:
Regarding integration test:
I have a bit of pain with having a test, that is run automatically but does not fully control its setup but instead tests against an external instance of an API. Maybe it is also just a matter of naming, as I know integration tests be run under controlled environments. Do you by chance have a link or an example, why this is a good practice? 

But in general I wouldn't mind to keep a test doing exactly this, but I would try to emphasize in the test's name, that it is running against an uncontrolled ecxternal instance.

Reply:
Ah, I see, sorry for the confusion😀I might be a bit sloppy there, but as far as I know, all test naming is sloppy😀

For me, I do it usually like this:
* Unit tests just test the actual unit, with all dependencies mocked
* Integration tests check the feature with the actual integration to other components, e.g. with using a real database instead of an in memory storage

So, I'm just calling it an integration test because it "integrates" with the actual http API of OpenProject. We could also call it E2E test or something (although I'm using that usually for when the UI is involved, e.g. frontend / backend combined under test).
The alternative would be to spin up an OpenProject instance in Docker when testing, but that usually adds around 30 to 60 seconds to tests when you start spinning up Docker containers there and setting up a database.

But, long story short, I don't think it's really important - I just usually try to have as many tests as possible, but I suspect the chance of the OpenProject API changing in that regard is very small.

Reply:
Also, making a second comment about the `using` part😀

Since the `HttpClientFactory` handles the disposing part of the `HttpClient`, I thought manually disposing it should be avoided. `using` internally just calls `Dispose()` on the client once it's out of scope.

However, the problem that originally led to the introduction of the `HttpClientFactory` was that when you did `Dispose()` a `HttpClient`, the `InnerHandler` was disposed as well. This still caused the TCP port to be reserved for like 90 seconds or so, so in high-usage scenarios this could lead to port exhaustion due to no more free ports available. This likely won't be a problem for us, but I'm wondering if there's a special handling with calling `Dispose()` directly on a `HttpClient` from an `HttpClientFactory`, maybe the `InnerHandler` just wraps the `Dispose()` and prevents it from becoming a problem?

I'll check up on that😀

Reply:
![image](https://user-images.githubusercontent.com/10274404/122765434-ca908000-d2a0-11eb-9314-ae6289ed0f6a.png)

Just found that in the docs, so the `using` is fine for `HttpClient`😀 Learned something new today!

Reply:
Great, did you check the test I added as Integration test? Does it fit?